storybookjs/storybook
# CLI: Detach automigrate command from storybook init

Closes #22346 



## What I did

- Automigrate does not run as part of `storybook init` anymore
- init command automatically detects and adds a babel config if needed (similar to the missing-babelrc automigration, but automated with no prompts)
- old ESLint "no-anonymous-default-exports" logic is removed as it was legacy thing for CRA and it's handled in the Storybook ESLint plugin
- eslint plugin logic was improved to also support package.json settings
- eslint plugin logic was refactored and reused in two places: 
  1. automigrations (as it existed already) for users who already have Storybook and are upgrading
  2. init (new) where it will suggest users to install the ESLint plugin as part of initializing Storybook

<img width="586" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/718eeab8-96d7-4bf5-bfa4-b0a7586dd355">

Text formatting isn't amazing but we plan on revamping the CLI to use something else and more beautiful in the future, I guess we can cope with it for now!

## How to test

1. Build the CLI
2. Go to different projects that do not have Storybook (use sandboxes repo) and use the `storybook init` command in those:
  a. [react-webpack](https://github.com/storybookjs/sandboxes/tree/next/react-webpack/17-ts/before-storybook) - should add babel config file
  b. [nextjs](https://github.com/storybookjs/sandboxes/tree/next/nextjs/default-js/before-storybook) - should suggest ESLint plugin and change the eslint config file
  c. [cra](https://github.com/storybookjs/sandboxes/tree/next/cra/default-js/before-storybook) - should suggest ESLint plugin and change config in package.json

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
